mm2src/lp_stats.rs

```diff
@@ -0,0 +1,334 @@
+                },
+            };
+
+            for (peer_id, response) in get_versions_res {
+                let name = match peers_names.get(&peer_id.to_string()) {
+                    Some(n) => n.clone(),
+                    None => continue,
+                };
+
+                match response {
+                    PeerDecodedResponse::Ok(v) => {
+                        let node_version_stat = NodeVersionStat {
+                            name: name.clone(),
+                            version: Some(v.clone()),
+                            timestamp,
+                            error: None,
+                        };
+                        if let Err(e) = insert_node_version_stat_to_db(&ctx, node_version_stat) {
+                            log::error!("Error inserting node {} version {} into db: {}", name, v, e);
+                        };
+                    },
+                    PeerDecodedResponse::Err(e) => {
+                        log::error!(
+                            "Node {} responded to version request with error: {}",
+                            name.clone(),
+                            e.clone()
+                        );
+                        let node_version_stat = NodeVersionStat {
+                            name: name.clone(),
+                            version: None,
+                            timestamp,
+                            error: Some(e.clone()),
+                        };
+                        if let Err(e) = insert_node_version_stat_to_db(&ctx, node_version_stat) {
+                            log::error!("Error inserting node {} error into db: {}", name, e);
+                        };
+                    },
+                    PeerDecodedResponse::None => {
+                        log::debug!("Node {} did not respond to version request", name.clone());
+                        let node_version_stat = NodeVersionStat {
+                            name: name.clone(),
+                            version: None,
+                            timestamp,
+                            error: None,
+                        };
+                        if let Err(e) = insert_node_version_stat_to_db(&ctx, node_version_stat) {
+                            log::error!("Error inserting no response for node {} into db: {}", name, e);
+                        };
+                    },
+                }
+            }
+        }
+        Timer::sleep(interval).await;
+    }
+}
+
+#[derive(Debug, Display)]
+pub enum ParseAddressError {
+    #[display(fmt = "Address {} resolved to IPv6 which is not supported", _0)]
+    UnsupportedIPv6Address(String),
+    #[display(fmt = "Address {} to_socket_addrs empty iter", _0)]
+    EmptyIterator(String),
+    // error return for second string
+    #[display(fmt = "Couldn't resolve '{}' Address: {}", _0, _1)]
+    UnresolvedAddress(String, String),
+}
+
+fn addr_to_ipv4_string(address: &str) -> Result<String, MmError<ParseAddressError>> {
```

Comment:
There is the similar `seed_to_ipv4_string` function in `lp_native_dex.rs`.
How about renaming, refactoring, and using it here instead of this?

Reply:
Maybe `addr_to_ipv4_string` can be moved to `lp_network` too and be used in both `lp_stats` and `lp_native_dex` while removing  `seed_to_ipv4_string` all together